### PR TITLE
Feature WhiteList

### PR DESCRIPTION
**Pull request by njuettner**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -124,9 +125,9 @@ func main() {
 		emergencyLTM := app.Group("/api/ltms/:lbpair")
 		emergencyLTM.Use(ginoauth2.Auth(ginoauth2.UidCheck, *OAuth2Endpoint, emergencyusers))
 		{
-			emergencyLTM.PATCH("/blockips", client.LTMBlockIPPatch)
-			emergencyLTM.PATCH("/whiteips", client.LTMWhiteIPPatch)
-			emergencyLTM.DELETE("/blockips", client.LTMRemoveBlockIPPatch)
+			emergencyLTM.PATCH("/blacklist", client.LTMBlockIPPatch)
+			emergencyLTM.PATCH("/whitelist", client.LTMWhiteIPPatch)
+			emergencyLTM.DELETE("/blacklist", client.LTMRemoveBlockIPPatch)
```
Comment (Raffo) [inline main.go]: There's no delete for whitelist?

**Comment by njuettner** [comment]
Its included now

File: client/client.go [hunk 2]
```diff
@@ -969,6 +969,24 @@ func LTMAddressList(c *gin.Context) {
...
 
+// LTMWhiteAddressList show local traffic whitelisted ip addresses
+func LTMWhiteAddressList(c *gin.Context) {
+	lbpair := c.Params.ByName("lbpair")
+	f5url, err := ltm.Loadbalancer(lbpair, common.Conf.Ltmdevicenames)
+	if err != nil {
+		respondWithStatus(err.Status, "", nil, err.Message, common.Conf.Documentation["ltmaddresslistdocumentationuri"], c)
+		return
+	}
+	f5url = strings.Replace(f5url, common.LtmURI, "", -1)
```
Comment (Raffo) [inline client/client.go]: Can we document the "magic" around "-1"? It would help the reader.

Comment (njuettner) [inline client/client.go]: Sure I'll comment it everywhere I've used it. IIRC its only used for network addresslists

File: client/client.go [hunk 3]
```diff
@@ -1040,13 +1058,38 @@ func LTMRemoveBlockIPPatch(c *gin.Context) {
...
+	var removewhiteips ltm.DeleteAddresses
+
+	f5url, err := ltm.Loadbalancer(c.Params.ByName("lbpair"), common.Conf.Ltmdevicenames)
+	if err != nil {
+		respondWithStatus(err.Status, "", nil, err.Message, common.Conf.Documentation["ltmaddresslistdocumentationuri"], c)
+		return
+	}
+	f5url = strings.Replace(f5url, common.LtmURI, "", -1)
+	if err := c.BindJSON(&removewhiteips); err != nil {
+		respondWithStatus(400, "Invalid JSON data", "Unblock IPs",
```
Comment (Raffo) [inline client/client.go]: Can you use http.StatusBadRequest?

File: client/client.go (hunk 2 above)
Comment (szuecs) [inline client/client.go]: Please use a constant, that describes what it means :)

File: config/config.go [hunk 4]
```diff
@@ -97,7 +97,7 @@ func LoadAuthConf(config *Config) ([]ginoauth2.AccessTuple, []ginoauth2.AccessTu
 	return rootUsers, emergencyUsers, &endpoint, nil
 }
 
-// LoadConfig load config file
+// init load config file
```
Comment (Raffo) [inline config/config.go]: Is this comment right? Golint would complain!

**Comment by Raffo** [comment]
I would say that the one above is my last comment, we can merge then. Sorry for the long wait!

**Pull request merged**